spec/deadlock/mutex_synchronize_spec.rb

```diff
@@ -0,0 +1,33 @@
+RSpec.describe Celluloid::Group do
+	class Worker
+		include Celluloid
+
+		def initialize
+			@mutex = Mutex.new
+		end
+
+		def wait
+			@mutex.synchronize do
+				sleep 3
+			end
+		end
+
+		def wait_for(worker)
+			worker.wait
+		end
+	end
+
+	unless RUBY_ENGINE == "ruby"
+		it "Mutex synchronize recovers from deadlock using Unlocker" do
```

Comment:
By the way, use `xit` or `pending` so that though the tests fail, their failure is allowed.